# Lattice of GaSb

In [86]:
from pymatgen.ext.matproj import MPRester
from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
from jupyter_jsmol.pymatgen import quick_view

In [87]:
with MPRester("SmhzvzkiqMXecCh0o4") as m:

    # Structure de mon cristal, GaSb
    structure = m.get_structure_by_material_id("mp-1018059")

s = SpacegroupAnalyzer(structure)

# Vecteurs de base du réseau direct

In [88]:
sg = s.get_conventional_standard_structure() # fonction qui renvoie les caractéristiques du réseau
direct = sg.lattice # fonction qui renvoie les vecteurs du réseau direct
print('Vecteurs de bases du réseau direct : \n'+str(sg.lattice) + '\n')
print('Paramètre de maille du réseau direct:')
print('a = '+str(direct.a))
print('b = '+str(direct.b))
print('c = '+str(direct.c))
print('\u03B1 = '+ str(direct.alpha))
print('\u03B2 = '+ str(direct.beta))
print('\u03B3 = '+ str(direct.gamma))

Vecteurs de bases du réseau direct : 
2.191967 -3.796599 0.000000
2.191967 3.796599 0.000000
0.000000 0.000000 7.207018

Paramètre de maille du réseau direct:
a = 4.383934340991306
b = 4.383934340991306
c = 7.207018
α = 90.0
β = 90.0
γ = 119.99999999999999


### Les vecteurs de base du réseau direct sont donnés par : 

* ${\bf a }= [2.191967;-3.796599;0.000000]$

* ${\bf b }= [2.191967;3.796599;0.000000]$

* ${\bf c }= [0.000000;0.000000;7.207018]$

### Les paramètres de maille du réseau direct vallent : 

* $ a = 4.383934340991306 Å$

* $ b = 4.383934340991306 Å$

* $ c = 7.207018 Å$

* $ \alpha = 90°$

* $ \beta = 90°$

* $ \gamma = 120°$

# Vecteurs de base du réseau réciproque



In [89]:

rl =direct.reciprocal_lattice # fonction qui renvoie les vecteurs du réseau réciproques


print('Vecteurs de bases du réseau réciproque : \n'+str(rl) + '\n')
print('Paramètres de maille du réseau réciproque:')
print('a* = '+str(rl.a))
print('b* = '+str(rl.b))
print('c* = '+str(rl.c))
print('\u03B1* = '+ str(rl.alpha))
print('\u03B2* = '+ str(rl.beta))
print('\u03B3* = '+ str(rl.gamma))

Vecteurs de bases du réseau réciproque : 
1.433230 -0.827476 0.000000
1.433230 0.827476 0.000000
0.000000 0.000000 0.871815

Paramètres de maille du réseau réciproque:
a* = 1.6549512133651865
b* = 1.6549512133651865
c* = 0.8718148486904829
α* = 90.0
β* = 90.0
γ* = 60.00000000000001


### Les vecteurs de base du réseau réciproque sont donnés par : 

* ${\bf a^* }= [1.433230;-0.827476;0.000000]$

* ${\bf b^* }= [1.433230;0.827476;0.000000]$

* ${\bf c^* }= [0.000000;0.000000;0.871815]$

### Les paramètres de maille du réseau réciproque vallent : 

* $ a^* = 1.6549512133651865 Å$

* $ b^* = 1.6549512133651865 Å$

* $ c^* = 0.8718148486904829 Å$

* $ \alpha^* = 90°$

* $ \beta^* = 90°$

* $ \gamma^* = 60°$

# Type de maille

In [90]:
x = s.get_lattice_type() #fonction qui renvoie le type de maille du cristal
print(x) 

hexagonal


La maille du cristal est hexagonale, maille qui est caractérisé par :
* $ a = b \ne c$
* $ \alpha = \beta = 90°$ et $\gamma = 120°$

# Système cristallin

In [91]:
x = s.get_crystal_system() #fonction qui renvoie le système cristallin du cristal
print(x)

hexagonal


La système du cristal est hexagonale, système qui est caractérisé par :
* $ a = b \ne c$
* $ \alpha = \beta = 90°$ et $\gamma = 120°$

# Groupe ponctuel

In [92]:
#groupe ponctuel
x = s.get_point_group_symbol() #fonction qui renvoie le groupe ponctuel du cristal
print(x)

6mm


Le groupe ponctuel du cristal est 6mm.